pr_reviewer/github_api.py

```diff
@@ -173,17 +173,120 @@ def get_pr_files(self, repo_name: str, pr_number: int) -> List[dict]:
             logger.error(f"❌ Failed to get PR files: {e}")
             raise Exception(f"Failed to fetch PR files: {str(e)}")
     
+    def extract_code_context(self, patch: str, target_line: int) -> dict:
+        """Extract original code context from a patch for generating suggestions"""
+        logger.debug(f"🔍 Extracting code context for line {target_line} from patch")
+        
+        if not patch:
+            return {"original_lines": [], "line_numbers": [], "context_start": 0}
+        
+        lines = patch.split('\n')
+        context = {
+            "original_lines": [],
+            "line_numbers": [],
+            "context_start": 0,
+            "hunk_info": None
+        }
+        
+        current_line_new = 0
+        current_line_old = 0
+        in_target_hunk = False
+        
+        for line in lines:
+            # Parse hunk headers (@@)
+            if line.startswith('@@'):
+                # Extract line numbers from hunk header: @@ -old_start,old_count +new_start,new_count @@
+                import re
+                match = re.match(r'@@ -(\d+),?\d* \+(\d+),?\d* @@', line)
+                if match:
+                    current_line_old = int(match.group(1))

```

Comment:
The regex pattern should be compiled once and reused for better performance

```suggestion
match = self._hunk_pattern.match(line)
```